### PR TITLE
TGUI chat improvements

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Tgui chat tab settings can now be toggled by clicking on a selected chat tab
Tgui chat settings now use a standard height of 175 pixels, instead of being inconsistent between menus
## Why It's Good For The Game
UI improvements

## Testing
Did both of these in game after running the dev server

## Changelog
:cl:
tweak: You can now open the tab settings by clicking on an already selected Tgui chat tab
tweak: the TGUI chat tab menu now uses a standard height of 175 pixels
/:cl:

### CODE REVIEW

**Review by Aylong220 — COMMENTED** [review]
Not sure about that... 
Can cause misunderstanding from players, plus it's a duplication of the settings button functionality

https://github.com/ParadiseSS13/Paradise/assets/69762909/500177f2-a58b-4d99-a796-6970d2f5a6c2

Also not sure about the size of the tab sections, the buttons on the home page have a large indentation at the bottom

image

**Comment by Aylong220** [comment]
> Not sure about that... Can cause misunderstanding from players, plus it's a duplication of the settings button functionality
> 
>  dreamseeker_PDvZZYvqMh.mp4 
> Also not sure about the size of the tab sections, the buttons on the general page have a large indentation at the bottom
> 
> image
> 
> Whereas now it's all fitted. image

And there's just enough space in the Text Highlight tab for 2 highlights
image

File: tgui/packages/tgui-panel/settings/SettingsPanel.js [hunk 1]
```diff
@@ -196,7 +196,7 @@ const TextHighlightSettings = (props, context) => {
   const highlightSettings = useSelector(context, selectHighlightSettings);
   const dispatch = useDispatch(context);
   return (
-    <Section fill scrollable height="230px">
```
Comment (S34NW) [inline tgui/packages/tgui-panel/settings/SettingsPanel.js:199]: I think we should go with the taller option, if anything, for these.

**Review by S34NW — COMMENTED** [review]
Unconvinced that clicking the current tab to open settings is needed when there is a big cog always there in the corner. Also see below regarding sizing.

**Comment by GDNgit** [comment]
> Unconvinced that clicking the current tab to open settings is needed when there is a big cog always there in the corner.

> Not sure about that... Can cause misunderstanding from players, plus it's a duplication of the settings button functionality

It enables the actual settings fully so it wouldn't cause any misunderstandings
Double click to edit is such a normal feature for UIs that I was instinctively doing this BEFORE I added this in. 

> Also not sure about the size of the tab sections, the buttons on the general page have a large indentation at the bottom
> Whereas now it's all fitted.

It's fitted at the cost of it being pretty awkward to use with the other chat menus. Filling that space isn't difficult either, Visual styles really should be buttons vs a dropdown menu considering we only have like six
> And there's just enough space in the Text Highlight tab for 2 highlights 

Yeah but if you're using two you're not using just two, you're using 4 or more. After four it becomes pretty busy and the regex menu feels pretty bloated. Maybe we could do something like changing the size of the menu depending on how many boxes are opened?

**Pull request closed without merging**